connector : dynamic channel by company

laetitia-gangloff opened the pull request:
On channel add "channel_by_company" flag to execute job in different channel in function of the company of the job

Use case :
Delivery Order (DO) are transfered by a job.
If many DO contains same product, the transfer failed.
For the moment, these job are retryable, but in some case (many DO) it is not sufficient.
So, we want use a channel with a capacity of 1.
But it is multi-company environment, so we want parallelize transfer of different companies.

sbidoul commented:
Hm. I understand the use case and the implementation looks fine.

However I wonder if it's not a bit too specific to go as is in the connector.

What would be of generic interest, though is a mecanism to let the developer select the channel at runtime when creating the job.

Commit:
connector : dynamic channel by company

On channel add "channel_by_company" flag to execute job in different channel in function of the company of the job

lmignon commented:
The generic approach is implemented by https://github.com/OCA/connector/pull/257
